src/Native/Runtime/portable.cpp

```diff
@@ -180,36 +181,161 @@ COOP_PINVOKE_HELPER(String *, RhNewString, (EEType * pArrayEEType, int numElemen
 
 #endif
 #if defined(USE_PORTABLE_HELPERS)
+struct RawEEType // TODO: defined in common.h
+{
+    uint16_t    m_componentSize;
+    uint16_t    m_flags;
+    uint32_t    m_baseSize;
+    MethodTable* m_pBaseType;
+    uint16_t    m_usNumVtableSlots;
+    uint16_t    m_usNumInterfaces;
+    uint32_t    m_uHashCode;
+};
 
-#ifdef HOST_ARM
+// dummy object for aligning next allocation to 8 that supports Methodtable GetBaseSize (12),HasComponentSize (false)
+static struct RawEEType dummy12ByteEEType = { 0, 0, 12, NULL, 0, 0, 0 };
+
+#if defined(HOST_ARM) || defined(HOST_WASM)
 COOP_PINVOKE_HELPER(Object *, RhpNewFinalizableAlign8, (EEType* pEEType))
 {
     Object * pObject = nullptr;
     /* TODO */ ASSERT_UNCONDITIONALLY("NYI");
```

Comment:
Haven't looked at this one, but if its trivial I could fill it out.

Reply:
Yes, this can be pretty simple - similar to `RhpNewFastMisalign`

Reply:
Right, I've done it, not committed though as I'm having a hard time getting a test.  The only types that I can make have `alignment.AsInt == 8` at https://github.com/dotnet/corert/blob/127405d0d75b4d4f2b83d96cb996e4a23088a4df/src/Common/src/TypeSystem/Common/MetadataFieldLayoutAlgorithm.cs#L829 are structs so can't put finalizers on those.

I thought a class with a `long` would be enough
```
    class FinalizableClass
    {
        long l; // requires 8 alignment 
        ~FinalizableClass()
        {
            l = 1;
            if(l == 1) finalizeCalled = true;
        }
    }
```

Reply:
Good point. You are right - this is de-facto unreachable. It may be worth it to add a comment about it.